llvm/tools/llvm-reduce/ReducerWorkItem.cpp

```diff
@@ -769,7 +769,11 @@ void ReducerWorkItem::readBitcode(MemoryBufferRef Data, LLVMContext &Ctx,
 }
 
 void ReducerWorkItem::writeBitcode(raw_ostream &OutStream) const {
+  const bool ShouldPreserveUseListOrder = true;
+
   if (LTOInfo && LTOInfo->IsThinLTO && LTOInfo->EnableSplitLTOUnit) {
+    // FIXME: This should not depend on the pass manager. There are hidden
```

Comment:
It is necessary because the pass may mutate the module by splitting into regular and ThinLTO sub modules. See my comment on your other PR about how to provoke this splitting.